lib/board.rb

```diff
@@ -0,0 +1,43 @@
+class Board
+  attr_accessor :cells
+
+  def initialize()
+    reset!
+  end
+
+  def reset!
+    @cells = Array.new(9, " ")
+  end
+
+  def display
+    puts " #{cells[0]} | #{cells[1]} | #{cells[2]} "
+    puts "-----------"
+    puts " #{cells[3]} | #{cells[4]} | #{cells[5]} "
+    puts "-----------"
+    puts " #{cells[6]} | #{cells[7]} | #{cells[8]} "
+  end
+
+  def full?
+    cells.all?{|token| token == "X" || token == "O"}
+  end
+
+  def turn_count
+    cells.count{|token| token == "X" || token == "O"}
+  end
+
+  def valid_move?(input)
+    input.to_i.between?(1,9) && !taken?(input)
+  end
+
+  def update(input, player)
+    cells[input.to_i-1] = player.token
+  end
+
+  def position(input)
+    cells[input.to_i-1]
+  end
+
+  def taken?(input)
+    !(position(input) == " " || position(input) == "")
+  end
```

Comment:
We can simplify this method by distributing the negation. This might be a little confusing, but `!(a || b)` <=> `!a && !b` (this property is one of [DeMorgan's Laws](https://en.wikipedia.org/wiki/De_Morgan%27s_laws)).